### PR TITLE
Pre-existing Comp Provisions Review

### PR DESCRIPTION
**Pull request by cameron-freshworks**
*Issue #:* /bcgov/entity#5177

*Description of changes:*

* Added review text when there is changes to the pre-existing company provisions.

* Fixed draft bug, where state wasn't restoring.

* Added Scroll to top effect when transitioning to Summary Mode.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -458,7 +458,7 @@ export default class App extends Mixins(CommonMixin, DateMixin, FilingTemplateMi
         // If Summary Mode: Check validity, save and file else move into summary mode.
         this.isSummaryMode
           ? await this.validateApp()
-          : this.setSummaryMode(true)
+          : this.scrollToTop(document.getElementById('app')); this.setSummaryMode(true)
```
Comment (severinbeauvais) [inline src/App.vue:461]: Do both statements on this line get executed? It's not clear from the syntax.

I think this would be better within an actual if-then-else with curly braces.

File: src/components/Articles/CompanyProvisions.vue [hunk 2]
```diff
@@ -145,7 +145,6 @@ export default class CompanyProvisions extends Mixins(CommonMixin) {
   @Watch('provisionsRemoved')
   private onProvisionsRemovedPropValueChanged (): void {
     if (!this.haveChanges) {
-      this.originalProvisionsRemovedValue = this.provisionsRemoved
```
Comment (andrepestana-aot) [inline src/components/Articles/CompanyProvisions.vue:148]: Will the originalProvisionsRemovedValue always start with false?

Comment (cameron-freshworks) [reply src/components/Articles/CompanyProvisions.vue:148]: Well it's a property local to the component, so it was always going to be whatever the prop value is. 

So when you resume a draft with a `provisionsRemoved` value of `true` it was setting the original value too ALSO `true`. It wasn't really a baseline "original" value. 

That being said, i'm not sure if a company could start an alteration with a provisionsRemoved value of `true`. 
I'll ask business on this one.

File: src/App.vue (hunk 1 above)
Comment (cameron-freshworks) [reply src/App.vue:461]: They do, because of the semi-colon. Javascript madness!

Will refactor for readability tho!

**Review by severinbeauvais — APPROVED** [review]

File: src/App.vue (hunk 1 above)
Comment (cameron-freshworks) [reply src/App.vue:461]: Updated in next commit!

File: src/components/Articles/CompanyProvisions.vue (hunk 2 above)
Comment (cameron-freshworks) [reply src/components/Articles/CompanyProvisions.vue:148]: So yea, this was a really good question! It may involve a little more design but i believe this is handled in the back end. 

TLDR: `hasRestrictions` is a boolean that is passed in the `businessInfo`, which dictates whether we capture this boolean value ( resolved pre existing provisions ). 

If `hasRestrictions` is true, it would show the component and indicate the pre existing provisions have not been resolved. So it is safe to assume at the component level that the original value would be false if we are rendering the component. 

After filing, i would assume the back end would resolve the `hasRestrictions`.

Comment (andrepestana-aot) [reply src/components/Articles/CompanyProvisions.vue:148]: My concern is about editing draft alteration where that could be selected before.

I think it makes sense to 'original' in this.originalProvisionsRemovedValue because it is original value since the component was mounted as it is updated only if there was no changes.

Comment (cameron-freshworks) [reply src/components/Articles/CompanyProvisions.vue:148]: Oh i see. Well in a DRAFT alteration is was broken because it was setting the original value to `true` if the user selected the checkbox, then saved / resumed a draft.

Comment (cameron-freshworks) [reply src/components/Articles/CompanyProvisions.vue:148]: Thats what this fix is for, to not overwrite the default value ( always `false` in this case ). So it could identify a diff between the state value of `provisionsRemoved`.

Comment (andrepestana-aot) [reply src/components/Articles/CompanyProvisions.vue:148]: > Oh i see. Well in a DRAFT alteration is was broken because it was setting the original value to `true` if the user selected the checkbox, then saved / resumed a draft.

In this case, I think it is correct because what we call 'original' here is there original value since the user started to work on the alteration. Not original since it was previously filed.

Comment (andrepestana-aot) [reply src/components/Articles/CompanyProvisions.vue:148]: Ok, after we chat I got what you're saying. You're right! Thanks!

**Review by andrepestana-aot — APPROVED** [review]
👍
